flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/RuntimeMonitorHandler.java

```diff
@@ -89,19 +91,20 @@ public RuntimeMonitorHandler(
 	}
 
 	@Override
-	protected void respondAsLeader(ChannelHandlerContext ctx, Routed routed, JobManagerGateway jobManagerGateway) {
+	protected void respondAsLeader(ChannelHandlerContext ctx, RoutedRequest routedRequest, JobManagerGateway jobManagerGateway) {
 		CompletableFuture<FullHttpResponse> responseFuture;
+		RouteResult result = routedRequest.getRouteResult();
```

Comment:
I think if you do `RouteResult<?> result = routedRequest.getRouteResult();` you don't need the cast to `Set<String>` in lines 101 and 106.